From khali@linux-fr.org Sun Dec 18 07:53:21 2005
Date: Sun, 18 Dec 2005 16:51:55 +0100
From: Jean Delvare <khali@linux-fr.org>
To: Greg KH <greg@kroah.com>
Subject: [PATCH 07/11] i2c: update i2c_driver.command documentation
Message-Id: <20051218165155.7d51c135.khali@linux-fr.org>
Content-Disposition: inline; filename=i2c-update-command-documentation.patch

Document i2c_driver.command as being deprecated, and don't suggest an
empty implementation of this callback as it doesn't make any sense.

Signed-off-by: Jean Delvare <khali@linux-fr.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 Documentation/i2c/writing-clients |   10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

--- gregkh-2.6.orig/Documentation/i2c/writing-clients
+++ gregkh-2.6/Documentation/i2c/writing-clients
@@ -492,17 +492,13 @@ Note that some functions are marked by `
 by `__init_data'.  Hose functions and structures can be removed after
 kernel booting (or module loading) is completed.
 
+
 Command function
 ================
 
 A generic ioctl-like function call back is supported. You will seldom
-need this. You may even set it to NULL.
-
-  /* No commands defined */
-  int foo_command(struct i2c_client *client, unsigned int cmd, void *arg)
-  {
-    return 0;
-  }
+need this, and its use is deprecated anyway, so newer design should not
+use it. Set it to NULL.
 
 
 Sending and receiving
